portal-impl/test/unit/com/liferay/portal/util/PortalImplEscapeRedirectTest.java

```diff
@@ -129,6 +130,20 @@ public void testEscapeRedirectWithDomains() throws Exception {
 		};
 		_redirectURLSettingsImpl.securityMode = "domain";
 
+		// Allow request Host header
+
+		Assert.assertEquals(
+			"https://" + _HOSTNAME_PORTAL_DOMAIN + ":8080",
+			_portalImpl.escapeRedirect(
+				"https://" + _HOSTNAME_PORTAL_DOMAIN + ":8080"));
+
+		// Allow Virtual Hosts
+
+		Assert.assertEquals(
+			"https://" + _HOSTNAME_VIRTUAL_HOST + ":8080",
+			_portalImpl.escapeRedirect(
+				"https://" + _HOSTNAME_VIRTUAL_HOST + ":8080"));
+
```

Comment:
Grouped the new tests here, separated into two sections starting with the same word, just like Brian asked